service/frontend/workflow_handler.go

```diff
@@ -369,7 +369,9 @@ func (wh *WorkflowHandler) StartWorkflowExecution(ctx context.Context, request *
 	}
 
 	if request.GetRequestId() == "" {
-		return nil, errRequestIDNotSet
+		// For easy direct API use, we default the request ID here but expect all
+		// SDKs and other auto-retrying clients to set it
+		request.RequestId = uuid.New()
```

Comment:
Do you think it's worth doing this only for http?

Reply:
Doing any call-specific things for HTTP is not only a bit difficult, it violates the spirit of "generic HTTP interface over existing gRPC behavior". So far I have been able to avoid any call-specific HTTP behavior.